packages/editor/src/components/post-actions/actions.js

```diff
@@ -498,7 +498,30 @@ const renamePostAction = {
 	id: 'rename-post',
 	label: __( 'Rename' ),
 	isEligible( post ) {
-		return post.status !== 'trash';
+		if ( post.status === 'trash' ) {
+			return false;
+		}
+		// Templates, template parts and patterns have special checks for renaming.
+		if (
+			! [
+				TEMPLATE_POST_TYPE,
+				TEMPLATE_PART_POST_TYPE,
+				...Object.values( PATTERN_TYPES ),
+			].includes( post.type )
+		) {
+			return true;
+		}
+		// In the case of templates, we can only remove custom templates.
+		if ( post.type === TEMPLATE_POST_TYPE ) {
+			return isTemplateRemovable( post ) && post.is_custom;
+		}
+		// Make necessary checks for template parts and patterns.
+		const isTemplatePart = post.type === TEMPLATE_PART_POST_TYPE;
+		const isUserPattern = post.type === PATTERN_TYPES.user;
+		const isCustomPattern =
+			isUserPattern || ( isTemplatePart && post.isCustom );
+		const hasThemeFile = isTemplatePart && post.templatePart.has_theme_file;
```

Comment:
```suggestion
		const hasThemeFile = isTemplatePart && post.has_theme_file;
```

I think this was a typo. It triggers an error when I open the template part for editing.

Reply:
Good catch! It seems it wasn't a typo but we change the template part object in patterns list. Fixed [here](https://github.com/WordPress/gutenberg/pull/61857/commits/9c4b3cbc3837caf049da62f8248882488e04da3c).